routes/api/v1/favorites.js

```diff
@@ -54,5 +58,54 @@ router.post("/", function(req, res) {
     })
 
     .catch(error => res.send({error}));
-})
+});
+
+router.get("/", function(req, res) {
+  User.findOne({
+    where: {
+      api_key: req.body.api_key
+    },
+    include: [Location]
+  })
+    .then( user => {
+      if(!user){
+        res.sendStatus(401)
+      } else{
+        let favorites = [];
+        user.getLocations()
+          .then(locations => {
+            var userLocations = locations;
+            for (i = 0; i < locations.length; i++) {
+              let currentLocation = locations[i];
+              let forecastUrl = `https://api.darksky.net/forecast/${process.env.DARK_SKY_API_KEY}/${currentLocation.lat},${currentLocation.lng}?exclude=[minutely,alerts,flags]`;
+              fetch(forecastUrl)
+                .then(response => {
+                  return response.json()
+                })
+                .then(forecastData => {
+                  var favorite = new FavoriteFormat(currentLocation.cityState, forecastData)
+                  favorites.push(favorite);
+                  return favorites;
+                })
+                .then(favoriteLocations => {
+                  if (favorites.length == userLocations.length){
+                    res.setHeader("Content-Type", "application/json");
+                    res.status(200).send(JSON.stringify(favorites));
+                  }
+                })
+                .catch(error => console.log(error));
```

Comment:
These different errors get a bit hard to read for somebody new to your code. One suggestion is to use more explicit error names. Or if you don't want to change them, you could even just add a few comments to differentiate them, since JS seems more forgiving of comments than Ruby.

Reply:
Changed the error names to be more user friendly